File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy = ListNode()
        dummy.next = head
        a = b = dummy

        for _ in range(n + 1):
            a = a.next
        
        while a:
            b = b.next
            a = a.next
        
        b.next = b.next.next 
        return dummy.next

File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeNthFromEnd(head *ListNode, n int) *ListNode {
    var dum = &ListNode{Val:0, Next: nil}
    dum.Next = head
    var a = dum
    var b = dum

    for i := 0; i < n + 1; i++ {
        a = a.Next
    }

    for a != nil {
        b = b.Next
        a = a.Next
    }

    b.Next = b.Next.Next
    return dum.Next
}